lib/routes/password.js

```diff
@@ -142,56 +145,140 @@ module.exports = function (
                   }
                 )
-            }
-          )
-          .then(
-            function () {
-              // Notify all devices that the account has changed.
-              push.notifyUpdate(passwordChangeToken.uid, 'passwordChange')
+              }
+            )
+        }
 
-              return db.account(passwordChangeToken.uid)
-                .then(
-                  function (account) {
-                    return mailer.sendPasswordChangedNotification(
-                      account.email,
-                      {
-                        acceptLanguage: request.app.acceptLanguage
+        function notifyAccount() {
+          // Notify all devices that the account has changed.
+          push.notifyUpdate(passwordChangeToken.uid, 'passwordChange')
 
-                      }
-                    )
+          return db.account(passwordChangeToken.uid)
+            .then(
+              function (accountData) {
+                account = accountData
+                return mailer.sendPasswordChangedNotification(
+                  account.email,
+                  {
+                    acceptLanguage: request.app.acceptLanguage
                   }
                 )
-            }
-          )
-          .then(
-            function () {
-              return {}
-            }
-          )
-          .done(reply, reply)
+              }
+            )
+        }
+
+        function createSessionToken() {
+          // Create a sessionToken with the verification status of the current session
+          var sessionTokenOptions = {
+            uid: account.uid,
+            email: account.email,
+            emailCode: account.emailCode,
+            emailVerified: account.emailVerified,
+            verifierSetAt: account.verifierSetAt,
+            tokenVerificationId: verifiedStatus ? undefined : crypto.randomBytes(16)
+          }
+
+          return db.createSessionToken(sessionTokenOptions, request.headers['user-agent'])
+            .then(
+              function (result) {
+                sessionToken = result
+              }
+            )
+        }
+
+        function createKeyFetchToken() {
+          if (wantsKeys) {
+            return db.createKeyFetchToken({
+                uid: account.uid,
+                kA: account.kA,
+                wrapKb: wrapKb,
+                emailVerified: account.emailVerified
+              })
+              .then(
+                function (result) {
+                  keyFetchToken = result
+                }
+              )
+          }
+        }
+
+        function createResponse () {
+          var response = {
+            uid: sessionToken.uid.toString('hex'),
+            sessionToken: sessionToken.data.toString('hex'),
+            verified: sessionToken.emailVerified,
+            authAt: sessionToken.lastAuthAt()
+          }
+
+          if (wantsKeys) {
+            response.keyFetchToken = keyFetchToken.data.toString('hex')
+          }
```

Comment:
@shane-tomlinson, if my memory serves correct, this is because the content server requests keys as part of the re-login process after password change?